modules/features/referrals/src/main/kotlin/au/com/shiftyjelly/pocketcasts/referrals/ReferralsViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package au.com.shiftyjelly.pocketcasts.referrals
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.repositories.user.UserManager
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.reactive.asFlow
+
+@HiltViewModel
+class ReferralsViewModel @Inject constructor(
+    private val userManager: UserManager,
+    private val settings: Settings,
+) : ViewModel() {
+    private val _state: MutableStateFlow<UiState> = MutableStateFlow(UiState())
+    val state: StateFlow<UiState> = _state
+
+    init {
+        viewModelScope.launch {
+            combine(
+                userManager.getSignInState().asFlow(),
+                settings.playerOrUpNextBottomSheetState,
```

Comment:
Tooltip auto hides on switching between tabs and going to another screen, however, it was laid above the full-screen player. I control its visibility by observing `playerOrUpNextBottomSheetState`.

Reply:
At first I wasn't sure what to think of it and that it might become a bit hard to maintain across different screens. But after reviewing the code and thinking more about it I like this solution and I think it is clever.